packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -394,31 +443,69 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
 
     private onMouseMove(event: React.MouseEvent) {
         this.positionXMove = event.nativeEvent.offsetX;
+        this.isMouseLeave = false;
+        if (this.mouseIsDown && this.isRightClick){
+            this.updateSelectionZoom();
+        }
     }
 
-    private onKeyDown(key: React.KeyboardEvent) {
-        switch (key.key) {
-            case 'W':
-            case 'w': {
-                this.zoom(ZOOM_IN);
-                break;
-            }
-            case 'S':
-            case 's': {
-                this.zoom(ZOOM_OUT);
-                break;
-            }
-            case 'A':
-            case 'a':
-            case 'ArrowLeft': {
-                this.pan(PAN_LEFT);
-                break;
+    private onMouseLeave(event: React.MouseEvent) {
+        this.isMouseLeave = true;
+        if (this.isRightClick) {
+            if (this.getTimeX(event.nativeEvent.offsetX) < this.props.unitController.viewRange.start){
+                this.setState({
+                    zoomSelectionRangeEnd: this.props.unitController.viewRange.start
+                });
+            } else if (this.getTimeX(event.nativeEvent.offsetX) > this.props.unitController.viewRange.end) {
+                this.setState({
+                    zoomSelectionRangeEnd: this.props.unitController.viewRange.end
+                });
             }
-            case 'D':
-            case 'd':
-            case 'ArrowRight': {
-                this.pan(PAN_RIGHT);
-                break;
+        }
+    }
+
+    private updateSelectionZoom() {
```

Comment:
Since you already keep track of the mouse x position, you could use that to draw the selection range, then you only need to convert it to a time when applying the zoom selection.

This way you wouldn't need to update the zoom end time as you pan/zoom.